staging/src/k8s.io/api/node/v1alpha1/types.go

```diff
@@ -114,3 +114,74 @@ type RuntimeClassList struct {
 	// Items is a list of schema objects.
 	Items []RuntimeClass `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// +k8s:conversion-gen:explicit-from=net/url.Values
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeLogViewerOptions is the query options for a Node's logs REST call.
+type NodeLogViewerOptions struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// SinceTime is an RFC3339 timestamp from which to show logs.
+	// +optional
+	SinceTime *metav1.Time `json:"sinceTime,omitempty" protobuf:"bytes,1,opt,name=sinceTime"`
+
+	// UntilTime is an RFC3339 timestamp until which to show logs.
+	// +optional
+	UntilTime *metav1.Time `json:"untilTime,omitempty" protobuf:"bytes,2,opt,name=untilTime"`
+
+	// TailLines if set, return up to this many lines (not more than 100k) from the end of the log.
+	// Support for this is implementation specific and currently only available for Linux and Windows service logs.
+	// +optional
+	TailLines *int64 `json:"tailLines,omitempty" protobuf:"varint,3,opt,name=tailLines"`
+
+	// Path is used to retrieve the contents of a literal file relative to the node's logs directory.
+	// If both a path and a service heuristic are specified, the service heuristic is performed.
+	// +optional
+	Path string `json:"path,omitempty" protobuf:"bytes,4,opt,name=path"`
+
+	// Format is used to display service logs in an alternate format.
+	// Support for this is implementation specific. Currently supported only for service logs on Linux.
+	// Valid values are: short, cat, json, short-unix.
+	// TODO: Make this into an enum type
+	// +optional
+	Format string `json:"format,omitempty" protobuf:"bytes,5,opt,name=format"`
+
+	// Pattern filters log entries by the provided regex pattern. On Linux nodes, this pattern will be read as a PCRE2 regex,
+	// on Windows nodes it will be read as a PowerShell regex.
+	// On Linux, if the pattern is all lowercase, matching is case insensitive. Otherwise, matching is case sensitive.
+	// This can be overridden with the PatternCaseSensitive option, see below.
+	// Support for this is implementation specific and currently only available for Linux and Windows service logs.
```

Comment:
For all of these impl-specifics, we should spec what happens if the user specifies them and the implementation doesn't suport it. Is it ignored or is it an error?

Reply:
It will be ignored. I have called that out now.